### PR TITLE
Extract the embedded file from the scripts

### PR DESCRIPTION
**Pull request by juanje**
The CI tests scripts have some embedded files for different purposes (add new source, create a blueprint, create a virtual network, etc), but it's easier to maintain if they are as separated files.
This makes the scripts smaller and easier to understand.

### CODE REVIEW

**Comment by juanje** [comment]
@thrix I extracted the files and the scripts are now far easier to follow :smile: 

Tell me what do you think.
Thanks!

File: tests/ci/create-commit.sh [hunk 1]
```diff
@@ -7,15 +7,15 @@ ID=${ID:-}
...
+BLUEPRINT_FILE=${BLUEPRINT_FILE:-}
 
 # Set up variables.
 TEST_UUID=$(uuidgen)
 IMAGE_KEY="osbuild-composer-ostree-test-${TEST_UUID}"
 
 # Set up temporary files.
 TEMPDIR=$(mktemp -d)
-BLUEPRINT_FILE=${TEMPDIR}/blueprint.toml
-SOURCE_FILE=${TEMPDIR}/copr_neptune.toml
```
Comment (thrix) [inline tests/ci/create-commit.sh:18]: maybe `APPLICATION_SOURCE_FILE` or `NEPTUNE_SOURCE_FILE` would be better?

File: tests/ci/create-commit.sh [hunk 2]
```diff
@@ -103,267 +103,15 @@ build_image() {
...
-id = "copr_neptune"
-name = "copr_neptune"
-type = "yum-baseurl"
-url = "https://download.copr.fedorainfracloud.org/results/pingou/qtappmanager-fedora/epel-8-${ARCH}"
-check_gpg = false
-check_ssl = false
-system = false
-EOF
+# Set the correct arch for the the Neptune app source file.
+sed -i "s|@@ARCH@@|${ARCH}|g" "$SOURCE_FILE"
```
Comment (thrix) [inline tests/ci/create-commit.sh]: what about having the files as templates and use envsubst instead?

https://www.gnu.org/software/gettext/manual/html_node/envsubst-Invocation.html

**Review by thrix — COMMENTED** [review]
LGTM, just small comments, it looks a lot better with not hardcoded files!

One thing I wanted to discuss, looking closer on how the image is built, it is not very friendly for the intended usage on AWS, could we enable also `cloud-init` for the image, I believe it has to work for rpm-ostree based distros, as Fedora CoreOS cloud image uses it ...

File: tests/ci/create-commit.sh (hunk 2 above)
Comment (juanje) [reply tests/ci/create-commit.sh]: I'm use to this kind of templates, I haven't used `envsubst` before, I'll give it a try.
Thanks.

**Comment by juanje** [comment]
> LGTM, just small comments, it looks a lot better with not hardcoded files!

Yey! It does!

> One thing I wanted to discuss, looking closer on how the image is built, it is not very friendly for the intended usage on AWS, could we enable also `cloud-init` for the image, I believe it has to work for rpm-ostree based distros, as Fedora CoreOS cloud image uses it ...

@thrix  I'm not sure how compatible the "RHEL for Edge" images are with `could-init` and such. I think the priority here was having something we could install in a edge-like device (i.e. Raspberry Pi).
Anyway, I think the Image Builder team is working in new types of images, one is cloud image. I'm not sure when that feature will be released, but it would simplify a lot all this (we won't need to build the image and then, install the VM).

We can create a ticket for the could images, and decide there if we wait for the official images or we add some changes to this one. In that last case, I don't know what is missing for the image to be cloud friendly. Any pointers would be much appreciate :smiley:

File: tests/ci/create-commit.sh (hunk 1 above)
Comment (lmilbaum) [reply tests/ci/create-commit.sh:18]: Can you please explain the `cloud-init` suggestion? I am not following how that can work and what is the value.

**Comment by thrix** [comment]
> > LGTM, just small comments, it looks a lot better with not hardcoded files!
> 
> Yey! It does!
> 
> > One thing I wanted to discuss, looking closer on how the image is built, it is not very friendly for the intended usage on AWS, could we enable also `cloud-init` for the image, I believe it has to work for rpm-ostree based distros, as Fedora CoreOS cloud image uses it ...
> 
> @thrix I'm not sure how compatible the "RHEL for Edge" images are with `could-init` and such. I think the priority here was having something we could install in a edge-like device (i.e. Raspberry Pi).
> Anyway, I think the Image Builder team is working in new types of images, one is cloud image. I'm not sure when that feature will be released, but it would simplify a lot all this (we won't need to build the image and then, install the VM).

Yep, that is true, the "production" use case won't need cloud-init. I guess I am fine bundling in an ssh key for now, but we need to make sure it is a key which also our provisioner knows about ... 

Let's for now drop my idea of `cloud-init` ....

> 
> We can create a ticket for the could images, and decide there if we wait for the official images or we add some changes to this one. In that last case, I don't know what is missing for the image to be cloud friendly. Any pointers would be much appreciate

The thing is, for some parts of the testing we would like to use public clouds. I would just try to import this qcow2 into AWS and see what does not work, then file the ticket. Maybe we can solve most of the problems in the kickstart installation, we mostly know what workarounds AWS needs.

**Pull request merged**